game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -258,12 +255,6 @@ public Route getRoute(final Territory t1, final Territory t2, final Predicate<Te
     checkNotNull(t1);
     checkNotNull(t2);
 
-    if (t1.equals(t2)) {
-      return new Route(t1);
-    }
-    if (getNeighbors(t1, cond).contains(t2)) {
-      return new Route(t1, t2);
-    }
     return new RouteFinder(this, cond).findRoute(t1, t2).orElse(null);
```

Comment:
As a side note now that this isn't really wrapping any advanced logic anymore:
If a RouteFinder would pre-calculate or cache the condition evaluations per territory and per selected unit configuration.
This instance creation should be done in a class that actually uses it, so that the same values aren't calculated multiple times.
This could save some performance as well.